device/logger.go

```diff
@@ -19,15 +19,25 @@ const (
 	LogLevelDebug
 )
 
-type Logger struct {
-	Debug *log.Logger
-	Info  *log.Logger
-	Error *log.Logger
+var _ Logger = &logger{}
+
+type Logger interface {
```

Comment:
In general I like the direction of this. There are a couple of points I think are worth discussing:

1. Do all six methods pull their weight? A Debugf/Infof/Errorf would get the job done.
2. Why an interface, and not a struct with three func fields?

Reply:
It depends how you want to do things in the code. Fine with formatted only. It really depends how code is being used. Less methods means easier to implement the interface... 

Interface is so I could hook any other logger implementation. Like this:
```
var _ device.Logger = &zapLogger{}
type zapLogger struct {
	zap *zap.Logger
}

func New(zap *zap.Logger) *zapLogger {
	return &zapLogger{
		zap: zap,
	}
}

func (l *zapLogger) Debug(v ...interface{}) {
	l.zap.Sugar().Debug(v...)
}
...
```